### PR TITLE
fix(webpack): Configure webpack for version 4

### PR DESCRIPTION
**Pull request by thallada**
Fixes the Webpack configs so that they work on version 4.

### CODE REVIEW

File: config/webpack.common.config.js [hunk 1]
```diff
@@ -7,7 +7,6 @@ module.exports = {
     app: path.resolve(__dirname, '../src/index.jsx'),
   },
   output: {
-    filename: '[name].js',
```
Comment (thallada) [inline config/webpack.common.config.js]: This is the default so unneeded.

File: config/webpack.dev.config.js [hunk 2]
```diff
@@ -2,13 +2,12 @@
...
 const Merge = require('webpack-merge');
 const path = require('path');
-const webpack = require('webpack');
 const HtmlWebpackPlugin = require('html-webpack-plugin');
 
 const commonConfig = require('./webpack.common.config.js');
 
 module.exports = Merge.smart(commonConfig, {
-  devtool: 'cheap-module-eval-source-map',
+  mode: 'development',
```
Comment (thallada) [inline config/webpack.dev.config.js]: The `development` mode sets the `devtool` to "eval". If the source maps are too hard to read then we can change this. "eval" is just the fastest to generate.

File: config/webpack.prod.config.js [hunk 3]
```diff
@@ -69,38 +69,20 @@ module.exports = Merge.smart(commonConfig, {
...
-      name: 'manifest',
-      minChunks: Infinity,
-    }),
     // Writes the extracted CSS from each entry to a file in the output directory.
     new ExtractTextPlugin({
       filename: '[name].min.css',
       allChunks: true,
     }),
-    // Defines a global variable that all bundles can access.
-    new webpack.DefinePlugin({
```
Comment (thallada) [inline config/webpack.prod.config.js]: mode: 'production' does this by default.

File: config/webpack.prod.config.js [hunk 4]
```diff
@@ -69,38 +69,20 @@ module.exports = Merge.smart(commonConfig, {
...
-      'process.env': {
-        // Maps the env var NODE_ENV at compile time to a global var available to bundles at
-        // runtime.  Should be set to "production" so that JS code knows it is running in a
-        // production environment.
-        NODE_ENV: JSON.stringify(process.env.NODE_ENV),
-      },
-    }),
-    // Minifies the JS.
-    new webpack.optimize.UglifyJsPlugin({ sourceMap: true }),
```
Comment (thallada) [inline config/webpack.prod.config.js]: mode: 'production' also does this by default.

File: package.json [hunk 5]
```diff
@@ -44,7 +44,7 @@
     "css-loader": "^0.28.9",
     "enzyme": "^3.3.0",
     "eslint-config-edx": "^4.0.3",
-    "extract-text-webpack-plugin": "^3.0.2",
+    "extract-text-webpack-plugin": "next",
```
Comment (thallada) [inline package.json]: Unfortunately, extract-text-webpack-plugin's webpack 4 support is still in beta right now.

Comment (jaebradley) [reply package.json]: Can you add an issue to bump this when it's not in beta?

File: config/webpack.prod.config.js (hunk 4 above)
Comment (jaebradley) [reply config/webpack.prod.config.js]: I know somebody can just "read the docs" but do you mind adding a comment about things that are defaulted? One of the things that was hard to keep track of was all the default behaviors that Webpack has.

Comment (thallada) [reply config/webpack.prod.config.js]: The funny thing is that there currently no docs on Webpack 4 and what `mode: production` does by default besides [this medium post](https://medium.com/webpack/webpack-4-mode-and-optimization-5423a6bc597a) :( .

I can try to enumerate everything in a comment here.

File: package.json (hunk 5 above)
Comment (jaebradley) [reply package.json]: Just so we can link things for historical purposes... 

#12 (thanks @thallada !)

File: config/webpack.prod.config.js (hunk 4 above)
Comment (jaebradley) [reply config/webpack.prod.config.js]: alt-text

**Pull request merged**